Release/rst 4.5

mts299 opened the pull request:
# RST 4.5 

This PR includes the following changes: 

## Testing 
Please indicate what machine you are testing on and the version we need tests done on the following machines:
- [x] Mac OsX @mtwalach @billetd 
- [x] Debian Linux - Ubuntu, XUbuntu, Mint @ecbland @kevinkrieger @egthomas 
- [x] Other - CentOS, fedora, OpenSUSE @aburrell @mts299 

Test the various  binaries include fitacf to map files 
plotting north and south hemisphere 
and using the new sound library 

### Documentation
https://radar-software-toolkit-rst.readthedocs.io/en/release-rst_4.5/

## Deadline: Dec 1st, 2020
* This could  be increased due to AGU if people are too busy but please let us know

## Approvals 
Please approve this PR and make a comment on the OS you used to make others aware of what has been tested and who has tested it. 

To approve a PR go to files changed at the top then top right there is `start a review`  in green button hit that and leave a comment on the OS you tested with and what you tested and then select Approve.

ecbland commented:
Thanks for making the release branch @mts299 !!

Note that the proposed author list for this release will come ASAP

mts299 commented:
@ecbland I updated the  release  guidelines, I am also gonna  make a  release  PR template to help with

ecbland commented on codebase/superdarn/src.bin/tk/tool/make_raw.1.8/make_raw.c:
```diff
@@ -214,11 +221,30 @@ int main (int argc,char *argv[]) {
 
+  command[0]=0;
+  n=0;
+  for (int c=0; i<argc; i++) {
```
Silly mistake that crept in from #320 

```suggestion
  for (int c=0; c<argc; c++) {
```

ecbland commented:
Here's the proposed author list for this release based on our new [authorship guidelines](https://superdarn.github.io/dawg/reports/202009/software-authorship). 

**Thomas, E.G., Schmidt, M.T., Bland, E.C., Burrell, A.G., Reimer, A.S., Sterne, K.T., and Walach, M.-T.**

Emphasis was given to (1) author of major new feature in this release (the `snd` library), and (2) other significant contributions to writing/testing code. Other contributors are listed alphabetically. In the guidelines we agreed to keep people on the author list for ~2-3 releases after their last contribution. For this purpose we went back to September 2018 (when development of RST4.3 began following the release of RST4.2). 

Please get in touch with me or @ksterne if you have any questions/comments relating to this. This is the first time we have done the author list this way, so any feedback would be useful.

ecbland commented:
Just capturing the list of updates from the RST4.4.2 patch release that we abandoned (the draft patch release can now be deleted):

- Error (core dumped) when using make_fit -old on some files
- Plotting grid and map vectors in southern hemisphere
- Defaults representative FoVs to 75 range gates with 45 km range separation
- Update of DLM functions to read dat files

mts299 commented:
Looks good and compiles on OpenSuse. Do we want a tutorial in the read the docs for sounding data?

mts299 commented on codebase/superdarn/src.bin/tk/plot/time_plot.1.8/snd.c:
```diff
@@ -0,0 +1,108 @@
+#include "sndseek.h"
+#include "tplot.h"
+
+
```
```suggestion 
// TODO: Add comments tothe code
```

For best practice, you should comment your code

mts299 commented on codebase/superdarn/src.bin/tk/plot/time_plot.1.8/snd.h:
```diff
@@ -0,0 +1,14 @@
+/* snd.h
+   =====
+   Author: E.G.Thomas
+*/
```
copyright-disclaimer

mts299 commented on codebase/superdarn/src.bin/tk/tool/combine_snd.1.0/combine_snd.c:
```diff
@@ -0,0 +1,198 @@
+/* combine_snd.c
+   =============
```
copyright, disclaimer, and comments for the code

mts299 replied:
```suggestion
Copyright (C) 2020, Dartmouth College
Author: E.G.Thomas

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/combine_snd.1.0/doc/combine_snd.doc.xml:
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
```
Wanna add this to the read the docs?

mts299 commented on codebase/superdarn/src.bin/tk/tool/combine_snd.1.0/makefile:
```diff
@@ -0,0 +1,17 @@
+# Makefile for combine_snd
+# ========================
+# Author: E.G.Thomas
+# by E.G.Thomas
```
this doesn't make sense :p 
copyright/disclaimer

mts299 replied:
```suggestion
# Copyright (C) 2020, Dartmouth College
# Author: E.G.Thomas
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/make_snd.1.0/doc/make_snd.doc.xml:
```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
```
readthedocs? To keep with consistency?

mts299 commented on codebase/superdarn/src.bin/tk/tool/make_snd.1.0/make_snd.c:
```diff
@@ -0,0 +1,159 @@
+/* make_snd.c
+   ==========
+   Author: E.G.Thomas
```
copyright/disclaimer/code comments

mts299 replied:
```suggestion
Copyright (C) 2020, Dartmouth College
Author: E.G.Thomas

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/make_snd.1.0/makefile:
```diff
@@ -0,0 +1,17 @@
+# =====================
+# Author: E.G.Thomas
+# by E.G.Thomas
+#
```
copyright/disclaimer

mts299 commented on codebase/superdarn/src.bin/tk/tool/rtsnd.1.0/loginfo.h:
```diff
@@ -0,0 +1,7 @@
+/* loginfo.h
+   =========
+   Author: R.J.Barnes
```
Is this correct? We should also have the appropriate copyright and disclaimer here as well

mts299 replied:
```suggestion
   Author: R.J.Barnes

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

I don't know what to put here for copyright or institution... let alone a year.

mts299 reviewed (commented):
Major changes on including proper code documentation and copyright/disclaimer on the sounding libraries.

ecbland commented:
> @mts299  Do we want a tutorial in the read the docs for sounding data?

More documentation is always welcome, but I think it's unfair to ask @egthomas to add more documentation when we have already created the release branch. The `snd` library PRs included updates to the [API documentation](https://superdarn.github.io/rst/), so I think that is sufficient for this release. I've added this as a "to do" on the [documentation project](https://github.com/SuperDARN/rst/projects/5).

kevinkrieger commented:
I'm sorry I don't have a debian system installed, I have opensuse right now.

mts299 commented:
Whoops sorry, @kevinkrieger that was supposed to be @ksterne 

Resolved the comments on reading the docs. 

I added my modification lines on all the files I changed for the complex PR.

mtwalach commented:
I'll test it on Monday - I've put it in my diary so I don't forget ;)

mtwalach commented on .github/ISSUE_TEMPLATE/deprecation_template.md:
```diff
@@ -0,0 +1,19 @@
+## Replacement
+
+*Would anything replace this feature?*
+
```
```suggestion
## Backward compatibility

*Does this potentially influence any old file formats or old code? Does it change backward compatibility?*
```
Just a suggestion - maybe change the wording but it might be worth considering

mts299 replied:
``` suggestion 
## Potential Issues 

- [ ] Backwards compatibility - *Does this potentially influence any old file formats or old code? Does it change backward compatibility?*
- [ ] reproducibility  - *Does this affect reproducibility between various RST versions?*

*If this affects one or both of the possible issues how would you inform or ensure people are aware and why this is needed.*
```

How about this? You make a good point and I realized the reproducibility issue too.

mtwalach reviewed (commented):
I've successfully installed this on Mac OS Catalina 10.15.7.

Attached is the log file, so you can take a look at the hundreds of warnings I got (I don't think they are anything to worry about but my C knowledge is limited).
[superdarn-rst.build.20201130.1251.txt](https://github.com/SuperDARN/rst/files/5616680/superdarn-rst.build.20201130.1251.txt)

mts299 reviewed (commented):
@mtwalach

ksterne commented:
Hey @mts299, sorry maybe I'm missing something, but I didn't see where you got Krieger and I mixed up.  Is there something you're wanting me to look at?

mts299 commented:
@ksterne at the top of the main descriptions of this PR I was wondering if you could test this on Ubunutu and that is where I hit Kevin K tag :p

ksterne commented:
Ah!  I see it now. Looks like it's been checked, but I do want to give it a basic "does it compile" test on a different Ubuntu.  I'll see what I can get to but don't let it hold things up.

mts299 commented:
No worries, go ahead for testing... actually not sure why it is checked off... hmmm

ecbland commented:
> No worries, go ahead for testing... actually not sure why it is checked off... hmmm

@mts299 It's checked because I tested it on Mint :)

mts299 commented:
Deadline has passed, @ecbland @ksterne what would you like to do? 
I think me and @mtwalach have approved it. We also need to add in the corrected author list.

ksterne commented:
@ecbland, I think you had a small update on the author list to add?

ksterne commented on codebase/superdarn/src.bin/tk/plot/time_plot.1.8/snd.c:
```diff
@@ -0,0 +1,108 @@
+/* snd.c
+   =====
+   Author: E.G.Thomas
```
@egthomas, are you able to add a copyright to this?  Maybe @mts299, can you add a suggestion?  I agree that it's likely we need @egthomas to commit it in, or at least approve of it.

ecbland commented:
Regarding the hardware files: I don't think there have been any changes since the last release. The most recent hardware file updates were to DCE, DCN, SPS & JME, which were added/updated for the [v4.3.2 release](https://zenodo.org/record/3775981).
- [ ] Could someone please check that I haven't missed any updates?

mts299 commented on codebase/superdarn/src.bin/tk/tool/combine_snd.1.0/combine_snd.c:
```diff
@@ -0,0 +1,198 @@
+/* combine_snd.c
+   =============
+   Author: E.G.Thomas
```
```suggestion
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/combine_snd.1.0/makefile:
```diff
@@ -0,0 +1,17 @@
+# Makefile for combine_snd
+# ========================
+# Author: E.G.Thomas
```
```suggestion
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/make_snd.1.0/makefile:
```diff
@@ -0,0 +1,17 @@
+# Makefile for make_snd
+# =====================
+# Author: E.G.Thomas
+# by E.G.Thomas
```
```suggestion
# Copyright (C) 2020, Dartmouth College
# Author: E.G.Thomas
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/rtsnd.1.0/makefile:
```diff
@@ -0,0 +1,18 @@
+# Makefile for rtsnd
+# ==================
+# Author: E.G.Thomas
+# by E.G.Thomas
```
```suggestion
# Copyright (C) 2020, Dartmouth College
# Author: E.G.Thomas
# 
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
```

mts299 commented on codebase/superdarn/src.bin/tk/tool/rtsnd.1.0/rtsnd.c:
```diff
@@ -0,0 +1,400 @@
+/* rtsnd.c
+   =======
+   Author: E.G.Thomas
```
```suggestion
Copyright (C) 2020, Dartmouth College
Author: E.G.Thomas

```

mts299 commented on codebase/superdarn/src.bin/tk/tool/rtsnd.1.0/rtsnd.c:
```diff
@@ -0,0 +1,400 @@
+/* rtsnd.c
+   =======
+   Author: E.G.Thomas
+*/
+
+/*
```
```suggestion

```

mts299 commented on codebase/superdarn/src.bin/tk/tool/rtsnd.1.0/rtsnd.c:
```diff
@@ -0,0 +1,400 @@
+/*
+ LICENSE AND DISCLAIMER
+ 
+ Copyright (c) 2012 The Johns Hopkins University/Applied Physics Laboratory
```
```suggestion
```
Is this correct? or do we need to move upwards?

mts299 reviewed (commented):
I did a few @ksterne and @ecbland but not all. This is not my responsibility and rather time-consuming to do them in suggestion boxes in GitHub. 

Hopefully, @egthomas can do the rest.

mts299 commented on .github/ISSUE_TEMPLATE/feature_template.md:
```diff
@@ -0,0 +1,50 @@
+-  [ ] Is this a major change?
+-  [ ] Do you need help developing it? 
+-  [ ] Have you created a RST project (github project) to reflect the process of developing it? 
+
```
```suggestion
-  [ ] Does it follow GPL 3.0v License and has been copyrighted/authored to you? 
```

mts299 reviewed (commented):
maybe this works? Maybe not let me know

mts299 commented:
@emma @ksterne do we want to release this soon? Looks like people still have compiler issues with C99 code in C90 compilation.

mts299 commented:
Happy New Year! :fireworks: Hope everyone had a good break! Wondering where we are at on this Release?

ksterne commented:
(whoops, I thought I'd hit comment before commenting on #355.  Kept running on tangents here.)
Hey all, great to see this merged in!  Great work everyone and thank you for your paitence!  One thing that's a bit troubling to me is I didn't see any communications here leading up to this merge.  I know a bunch have been working behind the scenes on this (thank you again for your efforts!), but that leaves others in the dark on what is going on.  For instance, one thing that I'm confused on is that it looks like a tag was made first and then the release was made?  Does that seem to make sense?  I'm wondering what happened here so that we can further update the SuperDARN/dawg#5 PR.  I know there's some outstanding suggestions there now and some that need to be added (in reference to creating the draft release) if they are there already.

Thinking of the release checklist, where are we with completing it out?  I see @ecbland's taking up on sending out an e-mail.  Does anyone want to take up making the PR from `master` back to `develop`?

mts299 commented:
Hi Kevin,

I already merged master back to develop

Cheers,
Marina

On Wed, Jan 13, 2021 at 9:23 AM Kevin Sterne <notifications@github.com>
wrote:

> (whoops, I thought I'd hit comment before commenting on #355
> <https://github.com/SuperDARN/rst/issues/355>. Kept running on tangents
> here.)
> Hey all, great to see this merged in! Great work everyone and thank you
> for your paitence! One thing that's a bit troubling to me is I didn't see
> any communications here leading up to this merge. I know a bunch have been
> working behind the scenes on this (thank you again for your efforts!), but
> that leaves others in the dark on what is going on. For instance, one thing
> that I'm confused on is that it looks like a tag was made first and then
> the release was made? Does that seem to make sense? I'm wondering what
> happened here so that we can further update the SuperDARN/dawg#5
> <https://github.com/SuperDARN/dawg/pull/5> PR. I know there's some
> outstanding suggestions there now and some that need to be added (in
> reference to creating the draft release) if they are there already.
>
> Thinking of the release checklist, where are we with completing it out? I
> see @ecbland <https://github.com/ecbland>'s taking up on sending out an
> e-mail. Does anyone want to take up making the PR from master back to
> develop?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/SuperDARN/rst/pull/365#issuecomment-759519374>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABRX5UQWAEOKOCQIJLFWRVTSZW3GRANCNFSM4TZDJI2A>
> .
>

mts299 commented:
Already mered `master` into `develop`

ecbland commented:
> One thing that's a bit troubling to me is I didn't see any communications here leading up to this merge. I know a bunch have been working behind the scenes on this (thank you again for your efforts!), but that leaves others in the dark on what is going on.

Sorry everyone for not communicating here about the release. This is squarely my fault. Indeed there was some communication going on behind the scenes, but not everyone was included in that. Let's keep this in mind for next time, and I hope I haven't prevented anyone from testing the code.

> I'm wondering what happened here so that we can further update the SuperDARN/dawg#5 PR. 

Yeah, some of the checklist did not make sense so we need to revisit this.

Once again thanks to everyone who has contributed!

ksterne commented:
@mts299, for info for others, I didn't see any pull request to merge from `master` back to `develop` but I do seem commits into the `develop` branch that are recent.  Is there a new way to do this on github that I've missed?  This talks to on updating the release checklist.

mts299 commented:
Moving convo  to release guidelines
